### ISSUE TITLE
longer sequence length

### ISSUE THREAD

**Issue by hyunwoongko** [bytedance/lightseq#197]
Any plans to support longer sequence lengths? 
Currently, the maximum length of the Softmax kernel seems 512.

**Comment by Taka152** [comment]
Thanks for your suggestion, are you talking about lightseq.inference?

**Comment by hyunwoongko** [comment]
nope. training

**Comment by hyunwoongko** [comment]
I was going to use it for GPT2 pretraining. 
I set it to 2048.

**Comment by Taka152** [comment]
OK, we'll relax this limitation in a new PR.

**Comment by hyunwoongko** [comment]
@Jack47 It's great. Did you test it works well?

**Comment by Taka152** [comment]
> max sequence length: 8836, having patched these lines of code and it seems works: in `void launch_attn_softmax_bw`:
> 
> ```cuda-c++
>   } else if (to_len <= 2048) {
>     grid_dim.x = 128;
>     ker_attn_softmax<__half, 1024, 2><<<grid_dim, 1024, 0, stream>>>(
>         inp, attn_mask, from_len, to_len, mask_future);
>   } else if (to_len <= 4096) {
>     grid_dim.x = 256;
>     ker_attn_softmax<__half, 1024, 4><<<grid_dim, 1024, 0, stream>>>(
>         inp, attn_mask, from_len, to_len, mask_future);
>   } else if (to_len <= 8192) {
>     grid_dim.x = 512;
>     ker_attn_softmax<__half, 1024, 8><<<grid_dim, 1024, 0, stream>>>(
>         inp, attn_mask, from_len, to_len, mask_future);
>   } else if (to_len <= 16384) {
>     grid_dim.x = 1024;
>     ker_attn_softmax<__half, 1024, 8><<<grid_dim, 1024, 0, stream>>>(
>         inp, attn_mask, from_len, to_len, mask_future);
>   } 
> ```
> 
> and in `launch_attn_softmax_bw`:
> 
> ```cuda-c++
>   else if (softmax_len <= 4096)
>     ker_attn_softmax_bw<T, 128>
>         <<<grid_dim, block_dim, 0, stream>>>(out_grad, soft_inp, softmax_len);
>   else if (softmax_len <= 8192)
>     ker_attn_softmax_bw<T, 256>
>         <<<grid_dim, block_dim, 0, stream>>>(out_grad, soft_inp, softmax_len);
>   else if (softmax_len <= 16384)
>     ker_attn_softmax_bw<T, 512>
>         <<<grid_dim, block_dim, 0, stream>>>(out_grad, soft_inp, softmax_len);
> ```

Yes, that's the place to modify the length limit, and it can be tested [here](https://github.com/bytedance/lightseq/blob/aabce486f34bec28bfe0efbbda1a183d5a6a37ba/tests/test_ls_kernels.py#L729-L730).
